### ISSUE TITLE
emergency mode after using Sandisk Extreme Pro USB drive (tested on 4 Gb as well as 8Gb Pi 4)

### ISSUE THREAD

**Issue by albertjve** [maxnet/berryboot#676]
I have tried this on Pi 4 4Gb as well as a Pi 4 8 gb to use an Sandisk Extreme Pro USB drive (128 Gb). This is a USB 3.2 Solid State Flash drive (SCSI). I could not make it work. I end up in "emercency mode" every time. 

A success and a failure:
1 - Using berryboot-20201103 zip I can successfully use Berryboot on an SD card. Load an OS and boot it. Change it etc.
Remark: I can only choose 64 bit (not 32 bit) Debian images. The list of OS to choose from is far more limited (approx 6) than the 10+ I expected. Not the issue, but this was surprising to me.

2. When choosing the Sandisk USB stick to store the OS-es, and than loading RPi_OS_2020_Lite[64-bit] on it and leaving the SD card in the pi to boot from:
- I reboot and can choose this OS in the berryboot menu (= normal operation)
- after that, starting this Operating System is confirmed  (= normal operation)
- then the boot process  pauses than for 1min30  "A start job is running for /dev/disk/by-label/boot" and times out
- "Dependency failed for /boot"
- "Dependency failed for local file systems"
- numerous processes that start successfully, followed by 
- "You are in emergency mode."... (and no other possibility than to reboot)

I searched the web but did not find no description of such behavior yet. 
also I have tried various ways to format the USB drive, but to no avail. 
I hope to be able to use this very fast USB stick that I have bought for this purpose. What to do?
I could post the lsusb details for the drive if that would help.

**Comment by kofec** [comment]
I have the same issue

**Comment by AlbatroZz** [comment]
I have the same problem but i can boot into Ubuntu 20.10.

Berryboot build 20201103

Hardware:
Raspberry Pi 4 8Gb
Argon One M.2
Crucial MX500 250Gb

**Comment by andreaz70** [comment]
Same issue here on an iscsi install.

**Comment by sjc** [comment]
I ran into this issue, too. It seems to be related in part to this other issue https://github.com/maxnet/berryboot/issues/667

Ultimately, renaming the FAT32 partition on the SD Card (not the USB device) to "boot" (lowercase) allowed the RPi OS I had previously installed to boot successfully.

**Comment by albertjve** [comment]
@all: thanks for your comments and suggestiosn. 

I have checked the label of the SD card, and changed it from BOOT (Windows 10 explorer showed 'boot') to boot. It did not solve the issue.
However since Ubuntu indeed launches successfully (as mentioned by AlbatroZz), it seems more an issue with the image files than something to do with the label name. 

To continue on the images, I thought it was peculiar that only 64 bit images were included. Therefore my hypothesis is that this Berryboot image uses another repository for the images with only 64 bit images and/or that SCSI is not yet supported in the Debian 64 bit images, so the USB stick (or iSCSI) fails to launch.

I would like to try some other (32 bit) images (e.g. https://sourceforge.net/projects/berryboot-updated-images/), but will have to tweak something on the SD card. Any suggestions?

**Comment by albertjve** [comment]
I will follow these instructions and test a 32 Debian image for Pi:
https://www.berryterminal.com/doku.php/berryboot/adding_custom_distributions

**Comment by albertjve** [comment]
instead if creating an image myself, I have downloaded a berryboot img and put it on a basic USB stick (non SCSI).
I can't however select this image in the Add OS menu (I only see the same predefined choices).

Documentation says:'Put your SquashFS formatted image on a USB stick, go to the “Operating system installer”, hold down your mouse button over “Add OS” and select “Install from USB stick” '
This somehow doesn't work. I am stuck. 
I hope some can help adding other images. Sorry this goes a bit of topic.

**Comment by albertjve** [comment]
After setting up a Berryboot repository (shared network folder), I could successfully use the Berryboot 32 bit Buster image. It loads almost instantly with the SCSI flash drive. Therefore the issue is related to incompatibility of 64 bit Debian (berryboot) images with SCSI devices (imho).

case closed.

**Comment by albertjve** [comment]
Thanks to all for making clear this is a general issue. And special thanks to AlbatroZz for pointing out Ubuntu did load, indicating the issue lies within the images itself.

**Comment by AlbatroZz** [comment]
Yeah i can confirm that the image is faulty. I've just created a custom image of Raspberry Os 64bit and it boots without a problem.

**Comment by albertjve** [comment]
I ran into issues with the 32 bit Buster image I used. It updated/upgraded succesfully, but e.g. my mouse disappeared and the file explorer in the GUI stopped working. I decided to look into other multiboot options (e.g. GRUB) but since the non standard boot process of the Pi this is a small project beyond my capabilities and the effort I want to spend.

@AlbatroZz : Did you run into issues when using this image?

Based on my experience with the quite old (2019) 32 bit image, I wonder whether berryboot remains stable after all kinds of update/upgrades.

**Comment by ekcsolutions** [comment]
Hi everyone. Ihave been following this thread as i  have the same problem, which i think i have just solved.
I have been watching the boot loading & noted it was stalling at loading the key board. i use a wireless key board so i just plugin a usb KB & it has just loaded. i hope this is your issue to as it is simple to fix.
cheers

**Comment by albertjve** [comment]
> Hi everyone. Ihave been following this thread as i have the same problem, which i think i have just solved.
> I have been watching the boot loading & noted it was stalling at loading the key board. i use a wireless key board so i just plugin a usb KB & it has just loaded. i hope this is your issue to as it is simple to fix.
> cheers

I have only been using a USB keyboard and mouse. So it would help me. However it adds to my earlier concerns regarding stability.

I am exploring now another option to multiboot.

**Comment by NeonFux** [comment]
I have the same problem with the 64bit Debian image on USB3 / sata external hard drive.
Several months later and nobody can fix the images ?

**Comment by owl770** [comment]
I believe the problem with the OS not being able to find /dev/disk/by-label/boot is an issue with the image rather than Berryboot itself. You could try the suggestion in #684 (editing `/etc/fstab` inside the image).
